modules/c++/nitf/source/BufferedReader.cpp

```diff
@@ -140,9 +143,40 @@ bool BufferedReader::canSeekImpl() const
 
 nitf::Off BufferedReader::seekImpl(nitf::Off offset, int whence)
 {
-    const sys::Off_T newOffset = mFile.seekTo(offset, whence);
-    readNextBuffer();
-    return newOffset;
+    const nitf::Off bufferEnd = mFile.getCurrentOffset();
+    const nitf::Off bufferStart = bufferEnd - mBufferSize;
+
+    nitf::Off desiredPos;
+    switch (whence)
+    {
+    case sys::File::FROM_START:
+        desiredPos = offset;
+        break;
+    case sys::File::FROM_CURRENT:
+        desiredPos = bufferStart + mPosition + offset;
+        break;
+    case sys::File::FROM_END:
+        desiredPos = mFile.length() + offset;
```

Comment:
Should this be subtraction?

Reply:
It follows the `lseek()` convention (where it is added): http://man7.org/linux/man-pages/man2/lseek.2.html.  `nitf::Off` is signed, so they'd send in a negative number.

Yeah it's weird. :o)